int128/groovy-ssh
# Parallel execution of sessions

This changes sessions to be run in parallel.


### Steps to verify the spec
Following commands should be run in parallel.

```groovy
        ssh.run {
            session(ssh.remotes.testServer) {
                execute 'sleep 2'
            }
            session(ssh.remotes.testServer) {
                execute 'sleep 1'
            }
        }
```

```
2016-09-18 22:57:39.685 INFO  Connected to Remote1[localhost:22] (SSH-2.0-OpenSSH_6.9)
2016-09-18 22:57:39.689 INFO  Connected to Remote2[localhost:22] (SSH-2.0-OpenSSH_6.9)
2016-09-18 22:57:40.167 INFO  Started command Remote1#0: sleep 2
2016-09-18 22:57:40.167 INFO  Started command Remote2#1: sleep 1
2016-09-18 22:57:41.212 INFO  Success command Remote2#1: sleep 1
2016-09-18 22:57:41.228 INFO  Disconnected from Remote2[localhost:22]
2016-09-18 22:57:42.182 INFO  Success command Remote1#0: sleep 2
2016-09-18 22:57:42.184 INFO  Disconnected from Remote1[localhost:22]
```

### Backward compatibility
This breaks backward compatibility. See also docs.

1. method signature of `ssh.run()`
2. exception thrown if 2 or more sessions are given and error occurs
3. `executeBackground` is now same as `execute`.
